Nef_3/include/CGAL/Nef_3/polygon_mesh_to_nef_3.h

```diff
@@ -205,7 +205,7 @@ void polygon_mesh_to_nef_3(PolygonMesh& P, SNC_structure& S, FaceIndexMap fimap,
   Face_graph_index_adder<typename SNC_structure::Items,
                  PolygonMesh, SNC_structure,HalfedgeIndexMap> index_adder(P,himap);
 
-
+  S.reserve_sm_boundary_items(num_vertices(P));
```

Comment:
>I am also not so sure in which scenario there could be a use for reserve(n) where n would not be known at the construction of the map itself.

@lrineau So for example in the above, (maybe this can be moved further up the call stack?). Hovever I guess its mostly for developer convenience, which is why I think all the other hash_maps compatible with std::unordered_map  implement a `reserve` method.